systemtests/volsupervisor_test.go

```diff
@@ -1,12 +1,61 @@
 package systemtests
 
 import (
+	"fmt"
 	"strings"
 	"time"
 
+	log "github.com/Sirupsen/logrus"
 	. "gopkg.in/check.v1"
 )
 
+func (s *systemtestSuite) TestVolsupervisorSnapLockedVolume(c *C) {
+	if !cephDriver() {
+		c.Skip("Only ceph supports snapshots")
+		return
+	}
+
+	_, err := s.uploadIntent("policy1", "snaplockedvol")
+	c.Assert(err, IsNil)
+
+	volName := genRandomVolume()
+	fqVolName := fqVolume("policy1", volName)
+	c.Assert(s.createVolume("mon0", fqVolName, nil), IsNil) // locked volume
+
+	time.Sleep(4 * time.Second)
```

Comment:
So, the volume(locked - only one mount possible on the host; when one container mounts it, no one else can mount it) is actually not in use until its mounted in some container. Here I have created the volume with this [snapunlocked.json](https://github.com/yuva29/volplugin/blob/2734854e9243d5cdeb6b0b27bb34690e67891a5f/systemtests/testdata/ceph/snaplockedvol.json) policy (snaps are taken every 2 seconds).

At L25, the sleep of 4s (gives time for 2 snap creation) but the volume is locked and not mounted(Not in use) so there should be NO snaps. That's what is checked in the first iteration of the loop (snapCount == prevCount == 0). Hope this helps. Please let me know if I made it confusing